src/libsync/syncengine.h

```diff
@@ -78,6 +78,11 @@ class OWNCLOUDSYNC_EXPORT SyncEngine : public QObject
     [[nodiscard]] SyncOptions syncOptions() const { return _syncOptions; }
     [[nodiscard]] bool ignoreHiddenFiles() const { return _ignore_hidden_files; }
 
+    [[nodiscard]] const ProgressInfo &progressInfo() const
+    {
+        return *_progressInfo;
```

Comment:
this would possibly cause an issue when `_progressInfo` is not a valid pointer
from a quick check, I would assume that maybe it would be better to not use `const ProgressInfo&` and just use `ProgressInfo*` to avoid dereferencing a possibly `nullptr` pointer